docs/baseplate/events.rst

```diff
@@ -6,6 +6,46 @@
 Building Events
 ---------------
 
+Thrift Schema v2 Events
+~~~~~~~~~~~~~~~~~~~~~~~
+
+For modern Thrift-based events: import the event schemas into your project,
+instantiate and fill out an event object, and pass it into the queue::
+
+   import time
+   import uuid
+
+   from baseplate.events import EventQueue, serialize_v2_event
+
+   from event_schemas.event.ttypes import Event
+
+
+   def make_wsgi_app(app_config):
+      ...
+
+      queue = EventQueue("v2", event_serializer=serialize_v2_event)
+      baseplate.add_to_context("events_v2", queue)
+
+      ...
+
+
+   def my_handler(request):
+      event = Event(
+         source="baseplate",
+         action="test",
+         noun="baseplate",
+         client_timestamp=time.time() * 1000,
+         uuid=str(uuid.uuid4()),
```

Comment:
🔕 One nice thing about the legacy format is you don't have to worry about setting fields like `uuid` and `client_timestamp` that you want on every event in a specific format.  Is there some way we can work that into this?  Or does it make more sense to just have a snippet that we can tell people "if you want to use v2 events, here's something to help you get started"?

Reply:
Yeah, I agree. That's definitely a downside here. This makes me a lot more sympathetic to distributing thriftfiles as pre-built python packages so we can add convenience wrappers to the generated code. Could also add some helpers to baseplate that just expect the right struct to be passed in or have a base schema that's public. :/